cli/tests/integrations/helpers/common.py

```diff
@@ -295,15 +298,19 @@ def ssh_output(cmd):
     # ssh must run with stdin attached to a tty
     proc, master = popen_tty(cmd)
 
-    # wait for the ssh command to finish
-    proc.wait(30)
-    returncode = proc.returncode
+    if wait:
+        # wait for the ssh command to finish
+        proc.wait(30)
```

Comment:
Actually, I'm confused, is this commit modifying the same function as the previous commit? maybe we can just squash both changes into one? Why did you have to add the option for `wait` vs. `poll` at all? Under what conditions is `expected_returncode != None` which would trigger you to use `wait` instead of `poll`?